src/herder/TxSetFrame.h

```diff
@@ -27,70 +27,94 @@ class AbstractTxSetFrameForApply
 
     virtual int64_t getBaseFee(LedgerHeader const& lh) const = 0;
 
-    virtual Hash const& getContentsHash() = 0;
+    virtual Hash const& getContentsHash() const = 0;
 
     virtual Hash const& previousLedgerHash() const = 0;
 
     virtual size_t sizeTx() const = 0;
 
     virtual size_t sizeOp() const = 0;
 
-    virtual std::vector<TransactionFrameBasePtr> sortForApply() = 0;
-    virtual void toXDR(TransactionSet& set) = 0;
+    virtual std::vector<TransactionFrameBasePtr> getTxsInApplyOrder() const = 0;
+    virtual void toXDR(TransactionSet& set) const = 0;
 };
 
+// A wrapper for a set of transactions that maintains the hash order
+
 class TxSetFrame : public AbstractTxSetFrameForApply
 {
-    std::optional<Hash> mHash;
+    using AccountTransactionQueue = std::deque<TransactionFrameBasePtr>;
+    using Transactions = std::vector<TransactionFrameBasePtr>;
+
+    // std::optional<Hash> mHash;
+    Hash mHash;
```

Comment:
Can we make all the members `const` to make sure they're immutable?

Reply:
I'm using `const` members whenever can. Some members, e.g Hash, Transactions, needs to be computed and assigned.

Reply:
Fixed it based on Graydon's suggestion [below](https://github.com/stellar/stellar-core/pull/3395#discussion_r854563736). All members are `const` now.